### PR TITLE
MGMT-11303: Add a deploy-lvmo component

### PR DESCRIPTION
**Pull request by flaper87**
# Description
   
Add a deploy-lvmo component
    
The LVMO component is part of the main repo as it is used for all the edgecluster, SNO, deployments. LVMO will serve as the base storage layer for SNO and other components like Quay.

Fixes # (issue)

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

Pipeline run:

image

openshift-storage pods:
image

LVMCluster:

image

Bound PVC with test pod (block):

image

Bound PVC with test pod (file):

image

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Comment by flaper87** [comment]
The tests may fail since the `odf-lvm-operator` is not mirrored in all of the CI hubs (just flavio6)

File: deploy-lvmo/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        echo ">>>> Waiting for subscription and crd on: ${edgecluster}"
+        echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+        declare -a LVMOCRDS=("lvmclusters.lvm.topolvm.io" "lvmvolumegroupnodestatuses.lvm.topolvm.io" "lvmvolumegroups.lvm.topolvm.io")
+        for crd in ${LVMOCRDS[@]}; do
+            check_resource "crd" "${crd}" "Established" "openshift-local-storage" "${EDGE_KUBECONFIG}"
+        done
+
+        echo ">>>> Render and apply manifest to deploy LVMCluster"
+        echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>"
+        render_file manifests/04-LVMO-LVMOCluster.yaml
```
Comment (danielchg) [inline deploy-lvmo/deploy.sh:86]: seems that apply the manifest 04-LVMO-LVMOCluster.yaml is missing in here?

**Review by danielchg — COMMENTED** [review]
only one comment regarding a posible missin oc apply

File: Makefile [hunk 2]
```diff
@@ -96,6 +96,18 @@ build-edgecluster-compact-2nics:
 wait-for-hub-sno:
 	${PWD}/shared-utils/wait_for_sno_mco.sh &
 
+
+run-pipeline-task:
```
Comment (alknopfler) [inline Makefile:100]: I think you need to add to .PHONY line 22 the name

File: deploy-lvmo/deploy.sh (hunk 1 above)
Comment (alknopfler) [reply deploy-lvmo/deploy.sh:86]: I think the apply is on line 18 (envsubst <${SOURCE_FILE} | oc --kubeconfig=${EDGE_KUBECONFIG} apply -f -) 
So this is OK

File: pipelines/resources/deploy-ztp-edgeclusters-sno.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 apiVersion: tekton.dev/v1beta1
 kind: Pipeline
 metadata:
-  name: deploy-ztp-edgeclusters-sno
+  name: deploy-ztp-edgeclusters-sno-lvmo
```
Comment (alknopfler) [inline pipelines/resources/deploy-ztp-edgeclusters-sno.yaml:4]: all sno will be lvmo ???

**Review by alknopfler — CHANGES_REQUESTED** [review]

**Review by alknopfler — APPROVED** [review]
LGTM

**Review by iranzo — APPROVED** [review]
lgtm, let's add id and add the missing operator on the hubs

**Pull request merged**